### PR TITLE
feat: add "kzg" as a separate feature

### PR DESCRIPTION
**Pull request by DaniPopes**
Separate kzg from std feature

### CODE REVIEW

File: crates/revm/Cargo.toml [hunk 1]
```diff
@@ -39,6 +39,12 @@ criterion = "0.5"
 
 [features]
 default = ["std", "secp256k1"]
```
Comment (Evalir) [inline crates/revm/Cargo.toml:41]: should `kzg` be turned on by default?

File: crates/primitives/Cargo.toml [hunk 2]
```diff
@@ -86,6 +71,7 @@ serde = [
     "bytes/serde",
     "bitvec/serde",
     "bitflags/serde",
+    "c-kzg?/serde",
```
Comment (Evalir) [inline crates/primitives/Cargo.toml:74]: TIL!

**Review by Evalir — APPROVED** [review]
nice! this should unblock snafus regarding building the foundry docker img

**Comment by rakita** [comment]
> Separate kzg from std feature

Supporting, would just rename feature to `ckzg`, to tie the name with c-kzg lib.

**Review by mattsse — APPROVED** [review]
lgtm

File: crates/precompile/Cargo.toml [hunk 3]
```diff
@@ -33,11 +33,22 @@ hex = "0.4"
...
+    "sha3/std",
+    "c-kzg?/std",
+    "secp256k1?/std",
+]
+
+# The following features may not work on all platforms as they depend on C libraries.
+# Enables the KZG point evaluation precompile.
+c-kzg = ["dep:c-kzg", "revm-primitives/c-kzg"]
+
+# Use `secp256k1` as a faster alternative to `k256`.
```
Comment (rakita) [inline crates/precompile/Cargo.toml:53]:
```suggestion
# Use `secp256k1` as a faster alternative to `k256`.
# The problem that `secp256k1` has, is it fails to build for `wasm` target on Windows and Mac as it is c lib.
# In Linux it passes. If you don't require to build wasm on win/mac, it is safe to use it and it is enabled by default.
```

**Pull request merged**